gateway/schema.go

```diff
@@ -0,0 +1,52 @@
+package gateway
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/getkin/kin-openapi/openapi3"
+
+	"github.com/TykTechnologies/tyk/apidef/oas"
+)
+
+type OASSchemaResponse struct {
+	Status  string      `json:"status"`
+	Message string      `json:"message"`
+	Schema  *openapi3.T `json:"schema,omitempty"`
+}
+
+func (gw *Gateway) schemaHandler(w http.ResponseWriter, r *http.Request) {
+	oasVersion := r.URL.Query().Get("oasVersion")
+
+	if oasVersion == "" {
+		doJSONWrite(w, http.StatusBadRequest, OASSchemaResponse{Message: "Should provide a value for parameter oasVersion", Status: "Failed"})
+		return
+	}
+
+	var resp OASSchemaResponse
+	var code = http.StatusOK
+
+	switch r.Method {
+	case http.MethodGet:
+
+		data := oas.GetOASSchema(oasVersion)
+		if data == nil {
+			resp.Message = fmt.Sprintf("Schema not found for version %s", oasVersion)
```

Comment:
oasVersion comes directly from Query().Get, use %q to quote/escape possibly malicious stuff (user generated content)